packages/graphql/src/scripts/transferActivitiesBetweenCourses.ts

```diff
@@ -0,0 +1,233 @@
+import { prisma } from '@klicker-uzh/prisma'
+import { recomputeDerivedPermissions } from '@klicker-uzh/util'
+import { EventEmitter } from 'events'
+
+// ! Script to transfer a set of activities between courses
+async function run() {
+  const emitter = new EventEmitter()
+
+  // ! Set old and new user IDs and the course IDs, which should be transferred including all activities and elements
+  const oldCourseId = ''
+  const newCourseId = ''
+  const liveQuizIds: string[] = []
+  const practiceQuizIds: string[] = []
+  const microLearningIds: string[] = []
+  const groupActivityIds: string[] = []
+  const chatbotIds: string[] = []
+
+  // fetch all content of the old user
+  const oldCourse = await prisma.course.findUnique({
+    where: { id: oldCourseId },
+    include: {
+      owner: true,
+      liveQuizzes: true,
+      practiceQuizzes: true,
+      microLearnings: true,
+      groupActivities: true,
+      chatbots: true,
+    },
+  })
+  const newCourse = await prisma.course.findUnique({
+    where: { id: newCourseId },
+    include: {
+      owner: true,
+    },
+  })
+
+  if (!oldCourse) {
+    throw new Error(`Origin course with ID ${oldCourseId} not found`)
+  }
+  if (!newCourse) {
+    throw new Error(`Destination course with ID ${newCourseId} not found`)
+  }
+
+  if (oldCourse.owner.id !== newCourse.owner.id) {
+    throw new Error(
+      `Old course owner and new course owner are not the same user (ID: ${oldCourse.ownerId})`
+    )
+  }
+  const oldUserId = oldCourse.owner.id
+  const newUserId = newCourse.owner.id
+
+  await prisma.$transaction(
```

Comment:
comment only: If the runtime of this transaction turns out to be a problem, I think we could also think about only wrapping the ownerId updates of the corresponding activities and their corresponding permission updates in a transaction, right?